kotest-assertions/kotest-assertions-core/src/commonMain/kotlin/io/kotest/matchers/collections/duplicates.kt

```diff
@@ -35,10 +36,20 @@ fun <T> Collection<T>.shouldNotContainDuplicates(): Collection<T> {
 }
 
 fun <T> containDuplicates() = object : Matcher<Collection<T>> {
-   override fun test(value: Collection<T>) = MatcherResult(
-      value.toSet().size < value.size,
-      { "Collection should contain duplicates" },
-      {
-         "Collection should not contain duplicates"
-      })
+   override fun test(value: Collection<T>): MatcherResult {
+      val duplicates = value.duplicates()
+      return MatcherResult(
+         duplicates.isNotEmpty(),
+         { "Collection should contain duplicates" },
+         {
+            "Collection should not contain duplicates, but has some: ${duplicates.print().value}"
+         })
+   }
 }
+
+internal fun<T> Collection<T>.duplicates(): Collection<T> = this.groupingBy { it }
+   .eachCount().entries
+   .mapNotNull { when(it.value) {
```

Comment:
Can you add a test for `List<T?>` with duplicated null entries? I wonder if this will detect that?

Reply:
> Can you add a test for `List<T?>` with duplicated null entries? I wonder if this will detect that?

done